src/backend/utils.ts

```diff
@@ -95,33 +95,33 @@ function semverGt(target: string, base: string) {
 
     // same major beta?
     if (bSplit[0] === tSplit[0]) {
-      base = bSplit[1]
-      target = tSplit[1]
+      base = bSplit[1] || ''
+      target = tSplit[1] || ''
       return target > base
     } else {
-      base = bSplit[0]
-      target = tSplit[0]
+      base = bSplit[0] || ''
+      target = tSplit[0] || ''
     }
   }
 
   // beta to stable
   if (base.includes('-beta')) {
-    base = base.split('-beta.')[0]
+    base = base.split('-beta.')[0] || ''
   }
 
   // stable to beta
   if (target.includes('-beta')) {
-    target = target.split('-beta.')[0]
+    target = target.split('-beta.')[0] || ''
   }
 
   const [bmajor, bminor, bpatch] = base.split('.').map(Number)
   const [tmajor, tminor, tpatch] = target.split('.').map(Number)
 
   let isGE = false
   // A pretty nice piece of logic if you ask me. :P
-  isGE ||= tmajor > bmajor
-  isGE ||= tmajor === bmajor && tminor > bminor
-  isGE ||= tmajor === bmajor && tminor === bminor && tpatch > bpatch
+  isGE ||= tmajor! > bmajor!
+  isGE ||= tmajor === bmajor && tminor! > bminor!
+  isGE ||= tmajor === bmajor && tminor === bminor && tpatch! > bpatch!
```

Comment:
I am not sure that this can not be undefined. We actually wrote a test for it where both passed values are undefined. I somehow don't like the `!` operator. With this you trust your code to much. Better are checks and tests.

Reply:
I'd argue that it would be the responsibility of the one calling this function to check the inputs first (in this case, making sure you're actually passing SemVer strings). But I agree, a simple check isn't too much to ask

Reply:
I would not take responsibility to the user. We working in a open source project where everyone have his design choices. I try to catch all possible errors, to avoid any traces and actually that should be our goal, else we never get stable. We got now 3 renderer traces causing empty screen in frontend.

Btw what happens if you still use the operator and write a test with both versions undefined. Does it return false?